mlflow/java/src/main/java/org/mlflow/sagemaker/ScoringServer.java

```diff
@@ -0,0 +1,282 @@
+package org.mlflow.sagemaker;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.util.thread.QueuedThreadPool;
+import org.mlflow.mleap.MLeapLoader;
+import org.mlflow.models.Model;
+
+/** A RESTful webserver for {@link Predictor Predictors} that runs on the local host */
+public class ScoringServer {
+  public static final String RESPONSE_KEY_ERROR_MESSAGE = "Error";
+
+  private static final int MINIMUM_SERVER_THREADS = 1;
+  // Assuming an 8 core machine with hyperthreading
+  private static final int MAXIMUM_SERVER_THREADS = 16;
```

Comment:
Some System.env if defined, else 16

Reply:
Done - also added a test: `testServerThreadConfigReadsMinMaxFromEnvironmentVariablesIfSpecified`. Unfortunately, Java doesn't provide support for writing environment variables, so this test depends on a mocked environment.